### PR TITLE
Allow markers only in their respective hemisphere when appropriate

### PR DESCRIPTION
**Pull request by vagechirkov**
#2331

### CODE REVIEW

File: nilearn/plotting/displays.py [hunk 1]
```diff
@@ -938,6 +938,22 @@ def add_markers(self, marker_coords, marker_color='r', marker_size=30,
             marker_coords_2d, third_d = _coords_3d_to_2d(
                 marker_coords, direction, return_direction=True)
             xdata, ydata = marker_coords_2d.T
+	        # Allow markers only in their respective hemisphere when appropriate
```
Comment (bthirion) [inline nilearn/plotting/displays.py]: Here you break the identation, leading to errors in the code. Can you fix it ?

Comment (bthirion) [reply nilearn/plotting/displays.py]: To make sure that this is right, include a pep8.flake8 checker in you Python editor.

Comment (vagechirkov) [reply nilearn/plotting/displays.py]: Yes, of course, sorry about this, I didn't notice

**Comment by kchawla-pi** [comment]
DO NOT Pull from master. There is some stuff that is going and until it has been attended to, do NOT pull from master.

**Comment by kchawla-pi** [comment]
Master is good now. All clear. You can go ahead and pull from it.

**Comment by NicolasGensollen** [comment]
Hi @vagechirkov, any updates on this? It looks like it was close to being merged...

**Comment by vagechirkov** [comment]
> Hi @vagechirkov, any updates on this? It looks like it was close to being merged...

Hi @NicolasGensollen, I would like to finish this, but I don't really understand what is missing.

**Comment by NicolasGensollen** [comment]
>Hi @NicolasGensollen, I would like to finish this, but I don't really understand what is missing.

I think the first step would be to merge master or to rebase on top of master such as to update the PR and enable the github action tests that were added recently. I can help with that if needed.

AFAICT, the review process ended up with a request from @bthirion for a non-regression test. @bthirion could you clarify what you meant ?

Note: Tests for `plot_glass_brain` are located in `nilearn/plotting/tests/test_img_plotting.py`

**Comment by bthirion** [comment]
@NicolasGensollen I'm not sure how to test it actually... Is there a way to assess that an axis only contains a glass brain outline ?

**Comment by NicolasGensollen** [comment]
@bthirion @vagechirkov In the last commit, I added a few tests to `test_add_markers_using_plot_glass_brain()` that might do the trick. Let me know what you think.

File: nilearn/plotting/tests/test_img_plotting.py [hunk 2]
```diff
@@ -1059,6 +1059,26 @@ def test_add_markers_using_plot_glass_brain():
     fig.add_markers(coords)
     fig.close()
 
+    # Add a single marker in right hemishpere such that no marker
+    # should appear in the left hemisphere when plotting
+    display = plotting.plot_glass_brain(None, display_mode='lyrz')
+    display.add_markers([[20,20,20]])
+    # Check that Axe 'l' has no marker
+    assert display.axes['l'].ax.collections[0].get_offsets().data.shape == (0,2)
```
Comment (bthirion) [inline nilearn/plotting/tests/test_img_plotting.py:1067]:
```suggestion
    assert display.axes['l'].ax.collections[0].get_offsets().data.shape == (0, 2)
``` 
(nitpick)

File: nilearn/plotting/tests/test_img_plotting.py [hunk 3]
```diff
@@ -1059,6 +1059,26 @@ def test_add_markers_using_plot_glass_brain():
...
+    # Check that all other Axes have one marker
+    for d in 'ryz':
+        assert display.axes[d].ax.collections[0].get_offsets().data.shape == (1,2)
+
+    # Add a single marker in left hemisphere such that no marker
+    # should appear in the right hemisphere when plotting
+    display = plotting.plot_glass_brain(None, display_mode='lyrz')
+    display.add_markers([[-20,20,20]])
```
Comment (bthirion) [inline nilearn/plotting/tests/test_img_plotting.py:1075]:
```suggestion
    display.add_markers([[-20, 20, 20]])
```

File: nilearn/plotting/tests/test_img_plotting.py [hunk 4]
```diff
@@ -1059,6 +1059,26 @@ def test_add_markers_using_plot_glass_brain():
...
+    # Check that Axe 'r' has no marker
+    assert display.axes['r'].ax.collections[0].get_offsets().data.shape == (0,2)
```
Comment (bthirion) [inline nilearn/plotting/tests/test_img_plotting.py:1077]: Same thing here

**Review by bthirion — COMMENTED** [review]
The test looks great, thx! 
Simple syntax nitpicks...

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Comment by NicolasGensollen** [comment]
Thanks for the PR @vagechirkov and for the reviews @bthirion !
Merging! :tada:

**Pull request merged**